### PR TITLE
Fix kafka plugin query failed when kafka key was raw formatted (long type), both projections and conditions contain kafka key

### PR DESCRIPTION
**Pull request by icanfly**
To resolve this issue:
https://github.com/prestodb/presto/issues/8364

### CODE REVIEW

File: presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java [hunk 1]
```diff
@@ -124,8 +124,22 @@ public RecordCursor cursor()
 
         private FieldValueProvider[] fieldValueProviders;
 
+        //cache values
```
Comment (losipiuk) [inline presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java]: Please explain in commit message what is fixed by this commit.

File: presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java [hunk 2]
```diff
@@ -236,17 +250,47 @@ private boolean nextRow(MessageAndOffset messageAndOffset)
                 }
             }
 
+            //clear cache
+            clearCaches();
+
             return true; // Advanced successfully.
         }
 
+        private void clearCaches()
```
Comment (losipiuk) [inline presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java]: use `Arrays.fill` instead for loops

**Comment by electrum** [comment]
Can you add a test case for this?

File: presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java (hunk 2 above)
Comment (icanfly) [inline presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java]: done

**Pull request closed without merging**